src/averageAges.js

```diff
@@ -14,13 +14,16 @@
  *
  * @return {number}
  */
-function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
-}
+const calculateMenAverageAge = (people, century) => {
+  const men = century ? people.filter(person =>
+    person.sex === 'm' && (Math.ceil(person.died / 100) === century))
+    : people.filter(person => person.sex === 'm');
```

Comment:
you just call filter once. and check the century inside
```suggestion
const men = people.filter(person => (
  century
    ? person.sex === 'm' && (Math.ceil(person.died / 100) === century)
    : person.sex === 'm'
));
```